tools/src/main/java/org/apache/kafka/tools/SchemaChecker/MetadataSchemaChecker.java

```diff
@@ -0,0 +1,347 @@
+
+                checkApiTypeVersions(jsonNode1, jsonNode2);
+                parser((ArrayNode) jsonNode1.get("fields"), (ArrayNode) jsonNode2.get("fields"));
+            }
+
+        } catch (FileNotFoundException e) {
+            System.out.println("An error occurred.");
+            e.printStackTrace();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static List<String> GetDataFromGit() throws IOException, GitAPIException {
+        List<String> gitSchemas = new ArrayList<>();
+
+        Git git = Git.open(new File(System.getProperty("user.dir") + "/.git"));
+        Repository repository = git.getRepository();
+        Ref head = git.getRepository().getRefDatabase().firstExactRef("refs/heads/trunk");
+
+        try (RevWalk revWalk = new RevWalk(repository)) {
+            RevCommit commit = revWalk.parseCommit(head.getObjectId());
+            RevTree tree = commit.getTree();
+            for (String jsonSchema : filesCheckMetadata) {
+                StringBuilder stringBuilder = new StringBuilder();
+                try (TreeWalk treeWalk = new TreeWalk(repository)) {
+                    treeWalk.addTree(tree);
+                    treeWalk.setRecursive(true);
+                    treeWalk.setFilter(PathFilter.create("metadata/src/main/resources/common/metadata/" + jsonSchema));
+                    if (!treeWalk.next()) {
+                        throw new IllegalStateException("Did not find expected file /metadata/src/main/resources/common/metadata/" + jsonSchema);
+                    }
+                    ObjectId objectId = treeWalk.getObjectId(0);
+                    ObjectLoader loader = repository.open(objectId);
+
+                    String content = new String(loader.getBytes());
+                    String[] lines = content.split("\n");
+                    boolean print = false;
+                    for (int i = 15; i < lines.length; i++) {
+                        if (lines[i].charAt(0) == '{') {
+                            print = true;
+                        }
+                        if (print && !lines[i].contains("//")) {
+                            stringBuilder.append(lines[i]);
+                        }
+                    }
+                }
+                gitSchemas.add(stringBuilder.toString());
+            }
+            revWalk.dispose();
+        }
+        return gitSchemas;
+    }
+
+    private static void checkApiTypeVersions(JsonNode original, JsonNode edited) {
+        if (!Objects.equals(original.get("apiKey"), edited.get("apiKey"))) {
+            throw new IllegalStateException("New schema has wrong api key, " + edited.get("apiKey") + " should be " + original.get("apiKey"));
+        }
+        if (!Objects.equals(original.get("type"), edited.get("type"))) {
+            throw new IllegalStateException("New schema has wrong record type, " + edited.get("type") + " should be " + original.get("type"));
```

Comment:
seems like "record types didn't match" is the problem ... we don't know which one is wrong